core/src/storage/impls/state.rs

```diff
@@ -265,6 +265,14 @@ impl<'a> StateTrait for State<'a> {
                 .free_owned_node(&mut owned_node.clone());
         }
     }
+
+    fn get_manifest(
```

Comment:
This file is not the correct place. Use snapshot_mpt.rs

Reply:
The state of checkpoint not only include the snapshot, but also intermediate and delta tries. On the other hand, user should not care about the concrete implementation of State (e.g. implemented by 3 tries). So, I defined the sync state interfaces in StateTrait.